### DISCUSSION TITLE
"Tracking" units through calculations

### DISCUSSION THREAD

**Discussion by rgilton** [KenKundert/quantiphy#31] [Q&A] [answered]
When I do a calculation like this:

```py
v = Quantity("1 A") * Quantity("10 kOhm")
print(v)
print(type(v))
```

I get:
```
10000.0
<class 'float'>
```

I would really like to end up with another `Quantity` instance that has units of Volts.  Is this possible with Quantiphy?

**Comment by KenKundert** [comment, marked as answer]
The whole unit propagation issue is a deep hole with many gotchas. It is outside the scope of QuantiPhy to address it.  There are other packages, such as pint, that specialize in units.  However it is possible to add unit propagation to QuantiPhy that works in typical cases when the problem is constrained.

Take a look at my [unit_propagation](https://github.com/KenKundert/unit_propagation) repository.  You can start with that and add rules (simplifications) to fit your situation.  The package is largely an experiment for me, so it is not in pypi.  It includes a simple RPN calculator demonstrator program that you can used to quickly get a feel for the capabilities and limitations of the package.
